doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
-\end{subfigure}
+  \caption{Eraser-fan commutation}
+  \begin{subfigure}[c]{0.1\textwidth}
+    \begin{tikzpicture}
+      \inetcell[]{a}{$f_i$}[5]
+      \inode[below = 40 pt]{b}{$⊗$}
+      \inwire[]{a/1}{a}
+      \inwire[]{a/4}{d}
+      \cutWire{a.out}{b.north}{}{}
+    \end{tikzpicture}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.05\textwidth}
+    \huge{→}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.1\textwidth}
+    \begin{tikzpicture}
+      \inode[]{a}{$⊗$}
+      \inode[right = 20 pt]{b}{$⊗$}
+      \inwire[]{a}{a}
+      \inwire[]{b}{d}
+    \end{tikzpicture}
+  \end{subfigure}
 \end{figure}
 
 \begin{figure}[H]
-\caption{Eraser-eraser annihilation}
-\begin{subfigure}[c]{0.1\textwidth}
-\begin{tikzpicture}
-\inode[]{a}{$⊗$}
-\inode[below = 40 pt]{b}{$⊗$}
-\vwire{a}{b}
-\end{tikzpicture}
-\end{subfigure}
-~
-\begin{subfigure}[c]{0.05\textwidth}
-\huge{→}
-\end{subfigure}
-~
-\begin{subfigure}[c]{0.1\textwidth}
-\end{subfigure}
+  \caption{Eraser-eraser annihilation}
+  \begin{subfigure}[c]{0.1\textwidth}
+    \begin{tikzpicture}
+      \inode[]{a}{$⊗$}
+      \inode[below = 40 pt]{b}{$⊗$}
+      \vwire{a}{b}
+    \end{tikzpicture}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.05\textwidth}
+    \huge{→}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.1\textwidth}
+  \end{subfigure}
 \end{figure}
 
 \floatstyle{boxed}
 \restylefloat{figure}
 
 ### Argument for correctness
 
+
 ## Bespoke term encoding
 
 - Primitive types (integer, string, bytes) ~> node types w/data
 - n-arity primitive functions (+, -, *, concat, index) ~> rewrite rules
-- alt-term encoding: $lam x. x + x + x$ ~> $3 * x$ rewrite rule instead of multi-node graph
+- alt-term encoding: $\lambda x. x + x + x$ ~> $3 * x$ rewrite rule instead of multi-node graph
 
 - Over set of underlying opcodes & von-Neumann-ish VM (e.g. x86, LLVM, EVM).
 - Use linear types (e.g. rewrite memory locations in place)
 - Must be semantically equivalent to interaction system encoding after read-back
 - Must preserve any-order reduction (exists some maximum `n` rewrite steps after which graph read-back must have final form regardless of order)
 - Can relax diamond property (strong confluence)? unclear if necessary
 
+### Argument for Correct read-back
+
+We wish to show that a read-back of an EAL-expression that has undergone zero or more reductions gives back the same result as normal evaluation up to alpha equality.
+
+Before we can prove this result, we must first prove a few lemmas and theorems first as well talk about what fan in's with the same label imply.
+
+The first lemma we wish to prove is the following
+
+\begin{Lemma}{AST→Net has one free port}{AST->Net}
+  Let $A$ = be a valid BOHM ADT.
+  \\ \\
+  Now Consider the net over $A$, let $N_a$ be this net.
+  \\ \\
+  Now let us consider any sub case of $A$, $L = P(n_1 \star \cdots \star n_k)$.
+  \\ \\
+  The node corresponding to $L$ looks.
+
+  \begin{tikzpicture}
+    \draw (0,0) circle (.8cm);
```

Comment:
This is a circle. I don't follow.

The sub-graph corresponding to the encoding of a subterm should be a graph with some number of free ports.

Reply:
This was the one unfinished graph, there is supposed to be n+1 ports, but alas wasn't sure how to draw that just yet